### DISCUSSION TITLE
If/else and how to select multiple without specifying amount?

### DISCUSSION THREAD

**Discussion by ShaneWikerSWE** [guidance-ai/guidance#109] [Q&A]
I'm trying to find how to do if x = 6, else if, etc.

Also how to have multiple selections of an array or object, without specifying how many to choose.

Like I may ask to select all of the items in an array being referenced or talked about, but without knowing exactly how many there are.

Is there a way to do that? The documentation and examples are limited so I haven't been able to figure it out

**Comment by PenutChen** [comment]
An example I came up with myself:
```python3
n = None


def set_bool_var(is_enabled):
    global n
    n = is_enabled == "true"


def my_cond():
    return n


prompt = """
JSON: { "enable": {{#select 'bool_var'}}true{{or}}false{{/select}} }{{set_bool_var bool_var}}
Result is {{#if my_cond}}True{{else}}False{{/if}}
"""

prog = guidance(prompt)

prog(set_bool_var=set_bool_var, my_cond=my_cond)
```

![image](https://github.com/microsoft/guidance/assets/94501378/4edfd6ee-780b-4ccb-8637-5a98ebcd9ace)


Some reference materials in the guidance/library/_if.py file might be helpful. Based on the comments, it appears that "else if" is currently not supported.

**Comment by slundberg** [comment]
`elif` support is something we would like to have, but don't yet. As for selecting more than one item from a select statement, you need to have a way for the llm to decide when it is done, which makes most sense as a list:
```python

guidance("""<items>{{#geneach 'selected_items' stop="</items>"~}}
<item>{{select 'this' options=all_options}}</item>{{/geneach}}</items>""")
```
